services/core-web/src/components/noticeOfWork/applications/NoticeOfWorkApplication.js

```diff
@@ -183,10 +183,11 @@ export class NoticeOfWorkApplication extends Component {
 
   loadNoticeOfWork = (id) => {
     this.props.fetchImportedNoticeOfWorkApplication(id).then(({ data }) => {
-      this.setState({ isLoaded: true });
       this.loadMineInfo(data.mine_guid);
       this.handleProgressButtonLabels(data.application_progress);
-      this.props.fetchOriginalNoticeOfWorkApplication(id);
+      this.props
+        .fetchOriginalNoticeOfWorkApplication(id)
+        .then(() => this.setState({ isLoaded: true }));
```

Comment:
The state isloaded should be probably happen after all of those promises resolve, is is that possible?

Reply:
Done. And I made it so the two calls that just need the NoW ID happen concurrently as well, rather than sequentially.